libraries/AP_HAL_ChibiOS/hwdef/Aerium-RadianF405/hwdef.dat

```diff
@@ -0,0 +1,204 @@
+# hw definition file for processing by chibios_pins.py
+# Omnibus F4 PRO with on-board current sensor
+# with F405 mcu, mpu6000 imu, bmp280 barometer, 7456 series osd  and sdcard
+
+MCU STM32F4xx STM32F405xx
+
+HAL_CHIBIOS_ARCH_F405 1
+
+# board ID for firmware load
+APJ_BOARD_ID TARGET_HW_AERIUM_RADIAN_F405
+
+# crystal frequency
+OSCILLATOR_HZ 8000000
+
+STM32_ST_USE_TIMER 4
+define CH_CFG_ST_RESOLUTION 16
+
+# flash size
+FLASH_SIZE_KB 1024
+FLASH_RESERVE_START_KB 64
+
+define STORAGE_FLASH_PAGE 1
+define HAL_STORAGE_SIZE 15360
+
+# spi1 bus for IMU
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+
+# spi2 for Barometer
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+# spi3 for SD-Card
+PC10 SPI3_SCK SPI3
+PC11 SPI3_MISO SPI3
+PC12 SPI3_MOSI SPI3
+
+# CS pins
+PB4 BMI270_CS CS
+PA4 LPS22HB_CS CS
+PB12 BMP280_CS CS
+PC13 SDCARD_CS CS
+PC3 ICM42688_CS CS
+
+# DRDY pins
+PC4 ICM42688P_DRDY INPUT
+PA15 BMI270_DRDY INPUT
+
+# SPI devices
+SPIDEV bmi270       SPI1 DEVID1 BMI270_CS  MODE3  10*MHZ  10*MHZ
+SPIDEV icm42688     SPI2 DEVID1 ICM42688_CS MODE3  2*MHZ  16*MHZ
+SPIDEV bmp280       SPI2 DEVID2 BMP280_CS MODE3 1*MHZ 8*MHZ
+SPIDEV lps22h       SPI1 DEVID2 LPS22HB_CS MODE3 1*MHZ 10*MHZ
+SPIDEV sdcard       SPI3 DEVID1 SDCARD_CS MODE0 400*KHZ 25*MHZ
+
+define HAL_OS_FATFS_IO 1
+define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
+define HAL_BOARD_TERRAIN_DIRECTORY "/APM/TERRAIN"
+
+# order of I2C buses
+I2C_ORDER I2C1 I2C2
+
+# order of UARTs
+SERIAL_ORDER OTG1 USART6 USART3 USART1
+
+#adc
+PC5 BATT_CURRENT_SENS ADC1 SCALE(1)
+PC0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+
+PC1 BATT2_CURRENT_SENS ADC1 SCALE(1)
+PC2 BATT2_VOLTAGE_SENS ADC1 SCALE(1)
+
+# pwm outputs
+PB1 TIM1_CH3N TIM1 PWM(1) GPIO(50)
+PB0 TIM1_CH2N TIM1 PWM(2) GPIO(51)
+PA8 TIM1_CH1  TIM1 PWM(3) GPIO(52)
+PA3 TIM2_CH4  TIM2 PWM(4) GPIO(53)
+PA2 TIM2_CH3  TIM2 PWM(5) GPIO(54)
+PA1 TIM2_CH2  TIM2 PWM(6) GPIO(55)
+PC9 TIM8_CH4  TIM8 PWM(7) GPIO(56)
+PC8 TIM8_CH3  TIM8 PWM(8) GPIO(57)
+
+# CAN Bus
+PB8 CAN1_RX CAN1
+PB9 CAN1_TX CAN1
+define CAN_APP_NODE_NAME "org.ardupilot.aerium"
+define UAVCAN_NODE_POOL_SIZE 1024
+
+# note that this board needs PULLUP on I2C pins
+PB6 I2C1_SCL I2C1
+PB7 I2C1_SDA I2C1
+
+# I2C2 for compass with BRD_ALT_CONFIG=1 These pins can also be used as USART3
+PB10 I2C2_SCL I2C2 ALT(1)
+PB11 I2C2_SDA I2C2 ALT(1)
+
+# USART3 with BRD_ALT_CONFIG=0
+PB10 USART3_TX USART3 ALT(0)
+PB11 USART3_RX USART3 ALT(0)
+
+PA10 USART1_RX USART1
+PA9 USART1_TX USART1
+
+PC6 USART6_TX USART6
+PC7 USART6_RX USART6
+
+# LEDs & safety
+PC15 LED_SAFETY OUTPUT
+PC14 SAFETY_IN INPUT PULLDOWN
+PB3 LED_BLUE OUTPUT HIGH GPIO(0)
+
+# Alarm
+PA0 TIM5_CH1 TIM5 GPIO(58) ALARM LOW
+define BUZZER_ENABLE_DEFAULT 0
+
+# USB
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# USB setup
+USB_VENDOR 0x1021
+USB_PRODUCT 0x1090
+USB_STRING_MANUFACTURER "Aerium"
```

Comment:
these defines will prevent firmware upload from automatically finding the right firmware for upload
please use the defaults